### PR TITLE
Schema updates for a new element

### PR DESCRIPTION
**Pull request by bpathak-ons**
### What

* Schema updates to include new topics (array) element received from zebedee
* Validation for the new element if does not exist in zebedee response

### How to review

* Code sense and CI pass

### Who can review

* Anyone

### CODE REVIEW

File: event/producer_test.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ var (
 		Summary:         "",
 		ReleaseDate:     "",
 		Title:           "",
+		Topics:          []string{""},
```
Comment (justinpjose) [inline event/producer_test.go:42]: I have a feeling that you were trying to initialize a string array with no elements. If that is the case, then please look at the suggestion. Right now, you have initialized a string array with 1 element which is an empty string

The same is done for `Keywords` as well

```suggestion
		Topics:          []string{},
```

File: models/mapper.go [hunk 2]
```diff
@@ -49,6 +53,14 @@ func RectifyKeywords(keywords []string, keywordsLimit int) []string {
 	return rectifiedKeywords[:keywordsLimit]
 }
 
+func ValidateTopics(topics []string) []string {
+	if topics == nil {
```
Comment (justinpjose) [inline models/mapper.go:57]: The `topics` array can either be a nil slice or an empty slice and currently, this is only checking if the array is nil. I believe this validation is also applicable when the array is empty as well

File: models/mapper.go [hunk 3]
```diff
@@ -49,6 +53,14 @@ func RectifyKeywords(keywords []string, keywordsLimit int) []string {
...
+		log.Warn(context.Background(), "warning : topics not present, i.e. null")
```
Comment (justinpjose) [inline models/mapper.go:58]: We shouldn't be creating another `context`. The `context` which is created when the service initially runs should be used. Therefore, this function should have a parameter called `ctx context.Context`

File: models/mapper.go [hunk 4]
```diff
@@ -49,6 +53,14 @@ func RectifyKeywords(keywords []string, keywordsLimit int) []string {
...
+		topics = []string{""}
```
Comment (justinpjose) [inline models/mapper.go:59]: As mentioned before, this is not an empty array
https://go.dev/play/p/n3N-WRSJolC

File: models/mapper_test.go [hunk 5]
```diff
@@ -170,6 +179,17 @@ func TestRectifyKeywords_EightKeywordsAndTenAsLimit(t *testing.T) {
 	})
 }
 
+func TestValidate_WithNiltopics(t *testing.T) {
```
Comment (justinpjose) [inline models/mapper_test.go:182]: You should also do another test case for when no topics are given (i.e when the array is empty - `len = 0`)

**Review by cookel2 — APPROVED** [review]
I ran it locally and I saw the empty list of topics in the log so it looks good.

**Pull request merged**